### DISCUSSION TITLE
Issue with useQuery

### DISCUSSION THREAD

**Discussion by peak-jiteshahuja** [TanStack/query#4221] [Q&A]
When using `useQuery`. For subsequent network calls with the same key `loading` state doesn't seem to reset. So basically when i change any of the keys being passed to `useQuery` the API call is re initiated but the `loading` state doesn't seem to change to true.

**Comment by TkDodo** [comment]
that is per design. `loading` means you have no data yet ...

**Comment by gpxl-dev** [comment]
@peak-jiteshahuja `isFetching` is `true` when fetching data, regardless of whether it's been fetched before.
